test/functional/interface_rest.py

```diff
@@ -455,6 +457,43 @@ def run_test(self):
                 expected = [(p["scriptPubKey"], p["value"]) for p in prevouts]
                 assert_equal(expected, actual)
 
+        self.log.info("Test the /blockpart URI")
+
+        blockhash = self.nodes[0].getbestblockhash()
+        block_bin = self.test_rest_request(f"/block/{blockhash}", req_type=ReqType.BIN, ret_type=RetType.BYTES)
+        for req_type in (ReqType.BIN, ReqType.HEX):
+            def _get_block_part(status: int = 200, **kwargs):
+                resp = self.test_rest_request(f"/blockpart/{blockhash}", status=status,
+                                              req_type=req_type, ret_type=RetType.BYTES, **kwargs)
+                assert isinstance(resp, bytes)
+                if req_type is ReqType.HEX and status == 200:
+                    resp = bytes.fromhex(resp.decode().strip())
+                return resp
+
+            assert_equal(block_bin, _get_block_part())
+            assert_equal(block_bin, _get_block_part(query_params={"offset": 0}))
+            assert_equal(block_bin, _get_block_part(query_params={"size": len(block_bin)}))
```

Comment:
I don't like these partial matches, is there any disadvantage in mandating both parameters?